registration/include/pcl/registration/impl/ndt.hpp

```diff
@@ -313,6 +348,48 @@ NormalDistributionsTransform<PointSource, PointTarget>::computePointDerivatives
 }
 
 
+template<typename PointSource, typename PointTarget> void
+NormalDistributionsTransform<PointSource, PointTarget>::computePointDerivatives (const Eigen::Vector3d &x, Eigen::Matrix<double, 4, 6>& point_jacobian, Eigen::Matrix<double, 24, 6>& point_hessian, bool compute_hessian) const
+{
+  // Calculate first derivative of Transformation Equation 6.17 w.r.t. transform vector.
+  // Derivative w.r.t. ith element of transform vector corresponds to column i, Equation 6.18 and 6.19 [Magnusson 2009]
+  Eigen::Matrix<double, 8, 1> point_angular_jacobian = angular_jacobian_ * Eigen::Vector4d(x[0], x[1], x[2], 0.0);
+  point_jacobian (1, 3) = point_angular_jacobian[0];
+  point_jacobian (2, 3) = point_angular_jacobian[1];
+  point_jacobian (0, 4) = point_angular_jacobian[2];
+  point_jacobian (1, 4) = point_angular_jacobian[3];
+  point_jacobian (2, 4) = point_angular_jacobian[4];
+  point_jacobian (0, 5) = point_angular_jacobian[5];
+  point_jacobian (1, 5) = point_angular_jacobian[6];
+  point_jacobian (2, 5) = point_angular_jacobian[7];
+
+  if (compute_hessian)
+  {
+    Eigen::Matrix<double, 15, 1> point_angular_hessian = angular_hessian_ * Eigen::Vector4d(x[0], x[1], x[2], 0.0);
+
+    // Vectors from Equation 6.21 [Magnusson 2009]
+    Eigen::Vector4d a(0, point_angular_hessian[0], point_angular_hessian[1], 0.0f);
+    Eigen::Vector4d b(0, point_angular_hessian[2], point_angular_hessian[3], 0.0f);
+    Eigen::Vector4d c(0, point_angular_hessian[4], point_angular_hessian[5], 0.0f);
+    Eigen::Vector4d d(point_angular_hessian[6], point_angular_hessian[7], point_angular_hessian[8], 0.0f);
+    Eigen::Vector4d e(point_angular_hessian[9], point_angular_hessian[10], point_angular_hessian[11], 0.0f);
+    Eigen::Vector4d f(point_angular_hessian[12], point_angular_hessian[13], point_angular_hessian[14], 0.0f);
+
+    // Calculate second derivative of Transformation Equation 6.17 w.r.t. transform vector.
+    // Derivative w.r.t. ith and jth elements of transform vector corresponds to the 3x1 block matrix starting at (3i,j), Equation 6.20 and 6.21 [Magnusson 2009]
+    point_hessian.block<4, 1>((9/3)*4, 3) = a;
+    point_hessian.block<4, 1>((12/3)*4, 3) = b;
+    point_hessian.block<4, 1>((15/3)*4, 3) = c;
+    point_hessian.block<4, 1>((9/3)*4, 4) = b;
+    point_hessian.block<4, 1>((12/3)*4, 4) = d;
+    point_hessian.block<4, 1>((15/3)*4, 4) = e;
+    point_hessian.block<4, 1>((9/3)*4, 5) = c;
+    point_hessian.block<4, 1>((12/3)*4, 5) = e;
+    point_hessian.block<4, 1>((15/3)*4, 5) = f;
+  }
+}
+
+// \note Thread-unsafe ```updateDerivatives``` is deprecated
```

Comment:
If it's possible with `computePointDerivatives`, something similar here too

Reply:
Done